### PR TITLE
Update Python client CHANGELOG

### LINKED ISSUE #1432: update python client's CHANGELOG
The [CHANGELOG.md file for the python client](https://github.com/cmu-delphi/delphi-epidata/blob/dev/src/client/packaging/pypi/CHANGELOG.md) is out of date.

As of this issue's creation (10 May 2024), it is missing:
* for version 4.1.13:
  * #1323 
  * #1330 
* for version 4.1.17:
  * #1363
* staged for release (likely to be in upcoming version 4.1.21):
  * #1418 

We should also update [the reference to the changelog in the package metadata](https://github.com/cmu-delphi/delphi-epidata/blob/a2dcfad838870ce05da45d4d3d9138e5cb60a36d/src/client/packaging/pypi/setup.py#L16) so that it points to the `main` branch instead of `dev`

### PR DESCRIPTION
**Pull request by rzats**
Closes #1432.

### Summary:

Updates the Python client's changelog to include items from previous releases.

Also includes a few currently pending changes. These can be omitted if we want to update the CHANGELOG right away; otherwise the date will need to be updated during the next release.

### Prerequisites:

- [x] Unless it is a documentation hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### CODE REVIEW

File: src/client/packaging/pypi/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,38 @@
...
+
+### Includes
+- https://github.com/cmu-delphi/delphi-epidata/pull/1323
+- https://github.com/cmu-delphi/delphi-epidata/pull/1330
+
+### Changed
+- Appends a trailing slash to URLs requested by the Python client, which should prevent an automatic redirect and an extra request to the server.
+
+### Removed
+- Mutes the covidcast_nowcast endpoint & removes unnecessary code that is related to this endpoint.
```
Comment (melange396) [inline src/client/packaging/pypi/CHANGELOG.md:36]: These notes should be phrased as being from the perspective of the client library and things its users will be interested in:
```suggestion
- Removed the `covidcast_nowcast()` method, as the associated API endpoint is no longer available.
```

File: src/client/packaging/pypi/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,38 @@
...
+
+### Includes
+- https://github.com/cmu-delphi/delphi-epidata/pull/1418
+- https://github.com/cmu-delphi/delphi-epidata/pull/1436
+
+### Added
+- Adds two debug flags: `debug` logs info about HTTP requests and responses; whereas `sandbox` prevents any HTTP requests from actually executing, allowing for tests that do not incur server load.
+
+### Changed
+- Updates internal version detection to use `bumpversion`.
```
Comment (melange396) [inline src/client/packaging/pypi/CHANGELOG.md:16]: Lets keep this PR focused on what is already committed; #1436 should make its own updates to the changelog.
```suggestion

### Added
- Adds two debug flags:
  - `debug` logs info about HTTP requests and responses
  - `sandbox` prevents any HTTP requests from actually executing, allowing for tests that do not incur server load.
```

File: src/client/packaging/pypi/CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,38 @@
...
+
+## [4.1.17] - 2024-01-30
+
+### Includes
+- https://github.com/cmu-delphi/delphi-epidata/pull/1363
+
+### Removed
+- Removes uses of setuptools' deprecated pkg_resources library - these are replaced by the native importlib.metadata library.
```
Comment (melange396) [inline src/client/packaging/pypi/CHANGELOG.md:24]: This was more a replacement than a removal
```suggestion
### Changed
- Replaced use of deprecated setuptools' `pkg_resources` library with the native `importlib.metadata` library.
```

**Review by melange396 — CHANGES_REQUESTED** [review]

**Comment by rzats** [comment]
@melange396 fixed!

File: src/client/packaging/pypi/CHANGELOG.md (hunk 2 above)
Comment (melange396) [reply src/client/packaging/pypi/CHANGELOG.md]: you took out the change from #1418!

Comment (rzats) [reply src/client/packaging/pypi/CHANGELOG.md]: added this in again!

**Review by dshemetov — APPROVED** [review]
Looked through the commits, seems like everything since Oct 2023 is included.

File: src/client/packaging/pypi/CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,31 @@
 All notable future changes to the `delphi_epidata` python client will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## [4.1.17] - 2024-01-30
```
Comment (melange396) [inline src/client/packaging/pypi/CHANGELOG.md:6]: you forgot to re-add the link to #1418 and that functionality should be under the pending 4.1.21 heading

**Review by melange396 — CHANGES_REQUESTED** [review]

**Review by melange396 — APPROVED** [review]

**Pull request merged**